Update docs for the 22.04 release[skip ci] (#4997)
NVIDIA/spark-rapids: docs/FAQ.md

### What versions of Apache Spark does the RAPIDS Accelerator for Apache Spark support?

> The RAPIDS Accelerator for Apache Spark requires version 3.1.1, 3.1.2, 3.1.3, 3.2.0 , 3.2.1 or 3.3.0

Reviewer: nit
```suggestion
The RAPIDS Accelerator for Apache Spark requires supports 3.1.1, 3.1.2, 3.1.3, 3.2.0 , 3.2.1 or 3.3.0
```

The RAPIDS Accelerator for Apache Spark officially supports:
- [Apache Spark](get-started/getting-started-on-prem.md)
- [AWS EMR 6.2+](get-started/getting-started-aws-emr.md)
- [Databricks Runtime 9.1, 10.4](get-started/getting-started-databricks.md)
- [Google Cloud Dataproc 2.0](get-started/getting-started-gcp.md)
- [Azure Synapse](get-started/getting-started-azure-synapse-analytics.md)

> - Cloudera provides Spark RAPIDS plugin packaged through 

Reviewer: Recommend we rewrite the cloudera one to be:
```
- [CDS 3.2](https://docs.cloudera.com/cdp-private-cloud-base/7.1.7/cds-3/topics/spark-spark-3-overview.html)
```
The requirements link is the second bullet point in the CDS documentation after overview, Ia m not sure we need to call it out.

Reviewer: I am fine with that. It is just different than the other bullet points, but that's minor.

Currently a limited set of SQL and DataFrame operations are supported, please see the
[configs](configs.md) and [supported operations](supported_ops.md) for a more complete list of what

> is supported. Some of the MLlib functions, such as `PCA` is supported.

Reviewer: ```suggestion
is supported. Some of the MLlib functions, such as `PCA` are supported.
```

finishes first then that's good, it is working as intended. If many tasks are speculating, but the
original task always finishes first then this is a pure loss, the speculation is adding load to
the Spark cluster with no benefit.

### Why is my query in GPU mode slower than CPU mode?

Below are some troubleshooting tips on GPU query performance issue:
* Make sure the query in GPU mode is fully on GPU. Please refer to

>   [Getting Started on Spark workload qualification](./get-started/getting-started-workload-qualification.md)

Reviewer 2: I think this is a bit too generic.. It the operator is only 2 ms for instance we may not care that everything is fully on GPU.  Also do we want to point to qualification tool or prefer the profiling tool and/or specifically look at the plan

Reviewer 2: that seems fine

  for more details. If there are some CPU fallbacks, check if those are some known features which
  can be enabled by turning on some Spark RAPIDS parameters. If the features needed do not exsit in
  the most recent release of Spark RAPIDS, please file 
  [feature request](https://github.com/NVIDIA/spark-rapids/issues) with a minimum reproduce.

* Tune the Spark and Spark RAPIDS parameters such as `spark.sql.shuffle.partitions`, 
  `spark.sql.files.maxPartitionBytes` and `spark.rapids.sql.concurrentGpuTasks` to get the best run.
  Please refer to [Tuning Guide](./tuning-guide.md) for more details.

* Identify the most time consuming part of the query. You can use 
  [Profiling tool](./spark-profiling-tool.md) to process the spark eventlog to get more insights of 

>   the query performance. For example, if IO is the bottleneck, we suggest optimize the backend 
>   storage IO performance. It's because the most suitable query type is computation bound instead of 
>   IO or Network bound.

Reviewer 3: ```suggestion
  the query performance. For example, if I/O is the bottleneck, we suggest optimizing the backend 
  storage I/O performance because the most suitable query type is computation bound instead of 
  I/O or network bound.
```

finishes first then that's good, it is working as intended. If many tasks are speculating, but the
original task always finishes first then this is a pure loss, the speculation is adding load to
the Spark cluster with no benefit.

### Why is my query in GPU mode slower than CPU mode?

Below are some troubleshooting tips on GPU query performance issue:

> * Identify the most time consuming part of the query. You can use

Reviewer: ```suggestion
* Identify the most time consuming part of the query. You can use the
```

>   [Profiling tool](./spark-profiling-tool.md) to process the Spark eventlog to get more insights of

Reviewer: ```suggestion
  [Profiling tool](./spark-profiling-tool.md) to process the Spark event log to get more insights of
```

  the query performance. For example, if I/O is the bottleneck, we suggest optimizing the backend
  storage I/O performance because the most suitable query type is computation bound instead of
  I/O or network bound.
  

> * Make sure at least the most time consuming part of the query is on GPU. Please refer to

Reviewer: ```suggestion
* Make sure at least the most time consuming part of the query is on the GPU. Please refer to
```

  [Getting Started on Spark workload qualification](./get-started/getting-started-workload-qualification.md)

>   for more details. Ideally we hope the whole query is fully on GPU, but if some minor part of the 

Author: @abellina sure i just changed it in FAQ guide for this section.

  query, eg. a small JDBC table scan, can not run on GPU, it won't cause much performance overhead.
  If there are some CPU fallbacks, check if those are some known features which can be enabled by 
  turning on some RAPIDS Accelerator parameters. If the features needed do not exist in the most 
  recent release of the RAPIDS Accelerator, please file a
  [feature request](https://github.com/NVIDIA/spark-rapids/issues) with a minimum reproduce.

* Tune the Spark and RAPIDS Accelerator parameters such as `spark.sql.shuffle.partitions`, 

>   `spark.sql.files.maxPartitionBytes` and `spark.rapids.sql.concurrentGpuTasks` to get the best run.

Reviewer: ```suggestion
  `spark.sql.files.maxPartitionBytes` and `spark.rapids.sql.concurrentGpuTasks` as these configurations have shown to affect performance of queries significantly.
```
